hook-common/src/webhook.rs

```diff
@@ -134,3 +134,22 @@ pub struct WebhookJobMetadata {
     pub plugin_id: Option<i32>,
     pub plugin_config_id: Option<i32>,
 }
+
+/// An error originating during a Webhook Job invocation.
+#[derive(Deserialize, Serialize, Debug, PartialEq, Clone)]
+pub struct WebhookJobError {
+    pub r#type: String,
+    pub details: String,
+    #[serde(with = "uuid::serde::compact")]
+    pub uuid: uuid::Uuid,
```

Comment:
Perhaps we should serialize this as a string...

Reply:
Need to add some tests to assert the behavior.

Reply:
Here's the `AppMetrics` types I made: https://github.com/PostHog/rusty-hook/pull/14/files#diff-a00f575cc51a6d6428477d58551304b1b9fefb596f6e9dd39f3e34870773e028